### PR TITLE
Requirement on LLVM >= 7 added on README #229

### LINKED ISSUE #229: cargo install failed
I tried to install grcov through cargo and received the following error:

```
error: failed to run custom build command for `grcov v0.3.0`
process didn't exit successfully: `/tmp/cargo-installZBuXsj/release/build/grcov-df85364459c28aad/build-script-build` (exit code: 101)
--- stdout
cargo:rustc-link-search=native=/usr/lib64

cargo:rustc-link-lib=static=LLVMProfileData
cargo:rustc-link-lib=static=LLVMCore
cargo:rustc-link-lib=static=LLVMBinaryFormat
cargo:rustc-link-lib=static=LLVMSupport
cargo:rustc-link-lib=static=LLVMDemangle
cargo:rustc-link-lib=dylib=z
cargo:rustc-link-lib=dylib=rt
cargo:rustc-link-lib=dylib=dl
cargo:rustc-link-lib=dylib=curses
cargo:rustc-link-lib=dylib=pthread
cargo:rustc-link-lib=dylib=m
cargo:rustc-link-lib=dylib=xml2
TARGET = Some("x86_64-unknown-linux-gnu")
OPT_LEVEL = Some("3")
HOST = Some("x86_64-unknown-linux-gnu")
CXX_x86_64-unknown-linux-gnu = None
CXX_x86_64_unknown_linux_gnu = None
HOST_CXX = None
CXX = None
CXXFLAGS_x86_64-unknown-linux-gnu = None
CXXFLAGS_x86_64_unknown_linux_gnu = None
HOST_CXXFLAGS = None
CXXFLAGS = None
DEBUG = Some("false")
running: "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I""/usr/include" "-Wall" "-Wextra" "-Wno-unused-parameter" "-fno-builtin" "-fno-exceptions" "-std=c++11" "-o" "/tmp/cargo-installZBuXsj/release/build/grcov-b76fd4146114cbf6/out/src/c/llvmgcov.o" "-c" "src/c/llvmgcov.cpp"
cargo:warning=src/c/llvmgcov.cpp: In member function ‘void CustomFileInfo::printIntermediate(llvm::StringRef, llvm::StringRef)’:
cargo:warning=src/c/llvmgcov.cpp:20:30: error: ‘std::__cxx11::string llvm::FileInfo::getCoveragePath(llvm::StringRef, llvm::StringRef)’ is private within this context
cargo:warning=   std::string CoveragePath = getCoveragePath(MainFilename, MainFilename);
cargo:warning=                              ^~~~~~~~~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:432:15: note: declared private here
cargo:warning=   std::string getCoveragePath(StringRef Filename, StringRef MainFilename);
cargo:warning=               ^~~~~~~~~~~~~~~
cargo:warning=src/c/llvmgcov.cpp:20:72: error: ‘std::__cxx11::string llvm::FileInfo::getCoveragePath(llvm::StringRef, llvm::StringRef)’ is private within this context
cargo:warning=   std::string CoveragePath = getCoveragePath(MainFilename, MainFilename);
cargo:warning=     ^
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:432:15: note: declared private here
cargo:warning=   std::string getCoveragePath(StringRef Filename, StringRef MainFilename);
cargo:warning=               ^~~~~~~~~~~~~~~
cargo:warning=src/c/llvmgcov.cpp:23:40: error: ‘std::unique_ptr<llvm::raw_ostream> llvm::FileInfo::openCoveragePath(llvm::StringRef)’ is private within this context
cargo:warning=   std::unique_ptr<raw_ostream> CovOs = openCoveragePath(FullCoveragePath);
cargo:warning=                                        ^~~~~~~~~~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:433:32: note: declared private here
cargo:warning=   std::unique_ptr<raw_ostream> openCoveragePath(StringRef CoveragePath);
cargo:warning=                                ^~~~~~~~~~~~~~~~
cargo:warning=src/c/llvmgcov.cpp:23:73: error: ‘std::unique_ptr<llvm::raw_ostream> llvm::FileInfo::openCoveragePath(llvm::StringRef)’ is private within this context
cargo:warning=   std::unique_ptr<raw_ostream> CovOs = openCoveragePath(FullCoveragePath);
cargo:warning=      ^
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:433:32: note: declared private here
cargo:warning=   std::unique_ptr<raw_ostream> openCoveragePath(StringRef CoveragePath);
cargo:warning=                                ^~~~~~~~~~~~~~~~
cargo:warning=src/c/llvmgcov.cpp: In instantiation of ‘void CustomFileInfo::printIntermediate(T&) [with T = llvm::GCOVOutputStream]’:
cargo:warning=src/c/llvmgcov.cpp:25:27:   required from here
cargo:warning=src/c/llvmgcov.cpp:41:7: error: ‘using FunctionLines = class llvm::DenseMap<unsigned int, llvm::SmallVector<const llvm::GCOVFunction*, 1> >’ is private within this context
cargo:warning=       FunctionLines::const_iterator FuncsIt = Line.Functions.find(LineIndex);
cargo:warning=       ^~~~~~~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:385:59: note: declared private here
cargo:warning=   using FunctionLines = DenseMap<uint32_t, FunctionVector>;
cargo:warning=                                                           ^
cargo:warning=src/c/llvmgcov.cpp:48:7: error: ‘using BlockLines = class llvm::DenseMap<unsigned int, llvm::SmallVector<const llvm::GCOVBlock*, 4> >’ is private within this context
cargo:warning=       BlockLines::const_iterator BlocksIt = Line.Blocks.find(LineIndex);
cargo:warning=       ^~~~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:387:53: note: declared private here
cargo:warning=   using BlockLines = DenseMap<uint32_t, BlockVector>;
cargo:warning=                                                     ^
cargo:warning=src/c/llvmgcov.cpp:53:15: error: ‘using BlockVector = class llvm::SmallVector<const llvm::GCOVBlock*, 4>’ is private within this context
cargo:warning=         const BlockVector &Blocks = BlocksIt->second;
cargo:warning=               ^~~~~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:386:56: note: declared private here
cargo:warning=   using BlockVector = SmallVector<const GCOVBlock *, 4>;
cargo:warning=                                                        ^
cargo:warning=src/c/llvmgcov.cpp:32:25: error: ‘llvm::StringMap<llvm::FileInfo::LineData> llvm::FileInfo::LineInfo’ is private within this context
cargo:warning=   for (const auto &LI : LineInfo)
cargo:warning=                         ^~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:447:23: note: declared private here
cargo:warning=   StringMap<LineData> LineInfo;
cargo:warning=                       ^~~~~~~~
cargo:warning=src/c/llvmgcov.cpp:39:28: error: ‘llvm::StringMap<llvm::FileInfo::LineData> llvm::FileInfo::LineInfo’ is private within this context
cargo:warning=     const LineData &Line = LineInfo[Filename];
cargo:warning=                            ^~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:447:23: note: declared private here
cargo:warning=   StringMap<LineData> LineInfo;
cargo:warning=                       ^~~~~~~~
cargo:warning=src/c/llvmgcov.cpp:64:13: error: ‘const llvm::GCOV::Options& llvm::FileInfo::Options’ is private within this context
cargo:warning=         if (Options.BranchInfo) {
cargo:warning=             ^~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:446:24: note: declared private here
cargo:warning=   const GCOV::Options &Options;
cargo:warning=                        ^~~~~~~
cargo:warning=src/c/llvmgcov.cpp: In instantiation of ‘void CustomFileInfo::printIntermediate(T&) [with T = llvm::GCOVOutputRust]’:
cargo:warning=src/c/llvmgcov.cpp:122:34:   required from here
cargo:warning=src/c/llvmgcov.cpp:41:7: error: ‘using FunctionLines = class llvm::DenseMap<unsigned int, llvm::SmallVector<const llvm::GCOVFunction*, 1> >’ is private within this context
cargo:warning=       FunctionLines::const_iterator FuncsIt = Line.Functions.find(LineIndex);
cargo:warning=       ^~~~~~~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:385:59: note: declared private here
cargo:warning=   using FunctionLines = DenseMap<uint32_t, FunctionVector>;
cargo:warning=                                                           ^
cargo:warning=src/c/llvmgcov.cpp:48:7: error: ‘using BlockLines = class llvm::DenseMap<unsigned int, llvm::SmallVector<const llvm::GCOVBlock*, 4> >’ is private within this context
cargo:warning=       BlockLines::const_iterator BlocksIt = Line.Blocks.find(LineIndex);
cargo:warning=       ^~~~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:387:53: note: declared private here
cargo:warning=   using BlockLines = DenseMap<uint32_t, BlockVector>;
cargo:warning=                                                     ^
cargo:warning=src/c/llvmgcov.cpp:53:15: error: ‘using BlockVector = class llvm::SmallVector<const llvm::GCOVBlock*, 4>’ is private within this context
cargo:warning=         const BlockVector &Blocks = BlocksIt->second;
cargo:warning=               ^~~~~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:386:56: note: declared private here
cargo:warning=   using BlockVector = SmallVector<const GCOVBlock *, 4>;
cargo:warning=                                                        ^
cargo:warning=src/c/llvmgcov.cpp:32:25: error: ‘llvm::StringMap<llvm::FileInfo::LineData> llvm::FileInfo::LineInfo’ is private within this context
cargo:warning=   for (const auto &LI : LineInfo)
cargo:warning=                         ^~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:447:23: note: declared private here
cargo:warning=   StringMap<LineData> LineInfo;
cargo:warning=                       ^~~~~~~~
cargo:warning=src/c/llvmgcov.cpp:39:28: error: ‘llvm::StringMap<llvm::FileInfo::LineData> llvm::FileInfo::LineInfo’ is private within this context
cargo:warning=     const LineData &Line = LineInfo[Filename];
cargo:warning=                            ^~~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:447:23: note: declared private here
cargo:warning=   StringMap<LineData> LineInfo;
cargo:warning=                       ^~~~~~~~
cargo:warning=src/c/llvmgcov.cpp:64:13: error: ‘const llvm::GCOV::Options& llvm::FileInfo::Options’ is private within this context
cargo:warning=         if (Options.BranchInfo) {
cargo:warning=             ^~~~~~~
cargo:warning=In file included from src/c/llvmgcov.cpp:4:
cargo:warning=/usr/include/llvm/ProfileData/GCOV.h:446:24: note: declared private here
cargo:warning=   const GCOV::Options &Options;
cargo:warning=                        ^~~~~~~
exit code: 1

--- stderr
thread 'main' panicked at '

Internal error occurred: Command "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/usr/include" "-Wall" "-Wextra" "-Wno-unused-parameter""-fno-builtin" "-fno-exceptions" "-std=c++11" "-o" "/tmp/cargo-installZBuXsj/release/build/grcov-b76fd4146114cbf6/out/src/c/llvmgcov.o" "-c" "src/c/llvmgcov.cpp"with args "c++" did not execute successfully (status code exit code: 1).

', /home/tristan957/.cargo/registry/src/github.com-1ecc6299db9ec823/cc-1.0.25/src/lib.rs:2260:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.

warning: build failed, waiting for other jobs to finish...
error: failed to compile `grcov v0.3.0`, intermediate artifacts can be found at `/tmp/cargo-installZBuXsj`

Caused by:
  build failed
```

### PR DESCRIPTION
**Pull request by LucianaMarques**
Fix #229

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -254,7 +254,7 @@ fn main() {
         let path_mapping_file = path_mapping_file.to_owned();
         let path_mapping = Arc::clone(&path_mapping);
 
-        thread::spawn(move || {
+        thread::Builder::new().name(String::from("Producer")).spawn(move || {
```
Comment (sylvestre) [inline src/main.rs]: This one is still marked as new ;)

Comment (marco-c) [reply src/main.rs]: @LucianaMarques try `git pull upstream master --rebase` and then `git push origin master --force`

Comment (LucianaMarques) [reply src/main.rs]: Ok, will do it today, sorry for the delay.

Comment (LucianaMarques) [reply src/main.rs]: `git pull upstream master --rebase
remote: Enumerating objects: 63, done.
remote: Counting objects: 100% (63/63), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 73 (delta 57), reused 60 (delta 57), pack-reused 10
Unpacking objects: 100% (73/73), done.
From https://github.com/mozilla/grcov
 * branch            master     -> FETCH_HEAD
 * [new branch]      master     -> upstream/master
First, rewinding head to replay your work on top of it...
Applying: naming threads as they are created
Using index info to reconstruct a base tree...
M	src/main.rs
.git/rebase-apply/patch:12: trailing whitespace.
            
warning: 1 line adds whitespace errors.
Falling back to patching base and 3-way merge...
Auto-merging src/main.rs
CONFLICT (content): Merge conflict in src/main.rs
error: Failed to merge in the changes.
Patch failed at 0001 naming threads as they are created
Use 'git am --show-current-patch' to see the failed patch

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort"`

And then when I try "git push origin master --force" it says it's already up-to-date

Comment (marco-c) [reply src/main.rs]: It seems like your repo is in a bad state, maybe try `git reset --hard upstream master` and then reapply your patch manually, then commit and `git push origin master --force`.

Comment (LucianaMarques) [reply src/main.rs]: Hi Marco, I'm sorry for not making this faster!

I'll certainly do it, sorry for the delay.

Comment (marco-c) [reply src/main.rs]: No need to apologize, you are helping us ;)

**Pull request closed without merging**